### ISSUE TITLE
IRI 1.4.1.4: NullPointerException while reading transaction.obsoleteTag

### ISSUE THREAD

**Issue by benoslav** [iotaledger/iri#424]
```
Error updating transactions.
Dez 20 23:25:25 server java[19036]: java.lang.NullPointerException: null
Dez 20 23:25:25 server java[19036]: at com.iota.iri.model.Transaction.metadata(Transaction.java:68) ~[iri-1.4.1.4.jar:na]
Dez 20 23:25:25 server java[19036]: at com.iota.iri.storage.rocksDB.RocksDBPersistenceProvider.update(RocksDBPersistenceProvider.java:385) ~[iri-1.4.1.4.jar:na]
Dez 20 23:25:25 server java[19036]: at com.iota.iri.storage.Tangle.update(Tangle.java:92) ~[iri-1.4.1.4.jar:na]
Dez 20 23:25:25 server java[19036]: at com.iota.iri.controllers.TransactionViewModel.update(TransactionViewModel.java:124) ~[iri-1.4.1.4.jar:na]
Dez 20 23:25:25 server java[19036]: at com.iota.iri.controllers.TransactionViewModel.updateHeights(TransactionViewModel.java:424) ~[iri-1.4.1.4.jar:na]
Dez 20 23:25:25 server java[19036]: at com.iota.iri.TransactionValidator.quickSetSolid(TransactionValidator.java:236) ~[iri-1.4.1.4.jar:na]
Dez 20 23:25:25 server java[19036]: at com.iota.iri.TransactionValidator.updateStatus(TransactionValidator.java:211) ~[iri-1.4.1.4.jar:na]
Dez 20 23:25:25 server java[19036]: at com.iota.iri.network.Node.processReceivedData(Node.java:407) [iri-1.4.1.4.jar:na]
Dez 20 23:25:25 server java[19036]: at com.iota.iri.network.Node.processReceivedDataFromQueue(Node.java:369) [iri-1.4.1.4.jar:na]
Dez 20 23:25:25 server java[19036]: at com.iota.iri.network.Node.lambda$spawnProcessReceivedThread$8(Node.java:579) [iri-1.4.1.4.jar:na]
Dez 20 23:25:25 server java[19036]: at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[na:1.8.0_151]
Dez 20 23:25:25 server java[19036]: at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[na:1.8.0_151]
Dez 20 23:25:25 server java[19036]: at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[na:1.8.0_151]
Dez 20 23:25:25 server java[19036]: at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[na:1.8.0_151]
Dez 20 23:25:25 server java[19036]: at java.lang.Thread.run(Thread.java:748) ~[na:1.8.0_151]
```

**Comment by alon-e** [comment]
additional case:
```
java.lang.NullPointerException: null
        at com.iota.iri.model.Transaction.metadata(Transaction.java:68) ~[IRI.jar:na]
        at com.iota.iri.storage.rocksDB.RocksDBPersistenceProvider.update(RocksDBPersistenceProvider.java:385) ~[IRI.jar:na]
        at com.iota.iri.storage.Tangle.update(Tangle.java:92) ~[IRI.jar:na]
        at com.iota.iri.controllers.TransactionViewModel.update(TransactionViewModel.java:124) ~[IRI.jar:na]
        at com.iota.iri.controllers.TransactionViewModel.setValidity(TransactionViewModel.java:313) ~[IRI.jar:na]
        at com.iota.iri.BundleValidator.validate(BundleValidator.java:105) ~[IRI.jar:na]
        at com.iota.iri.LedgerValidator.getLatestDiff(LedgerValidator.java:84) ~[IRI.jar:na]
        at com.iota.iri.LedgerValidator.updateSnapshot(LedgerValidator.java:249) ~[IRI.jar:na]
        at com.iota.iri.Milestone.updateLatestSolidSubtangleMilestone(Milestone.java:228) ~[IRI.jar:na]
        at com.iota.iri.Milestone.lambda$init$1(Milestone.java:150) ~[IRI.jar:na]
        at java.lang.Thread.run(Thread.java:748) ~[na:1.8.0_131]
12/19 17:03:04.912 [Solid Milestone Tracker] ERROR com.iota.iri.Milestone - Error during Solid Milestone updating
java.lang.NullPointerException: null
        at com.iota.iri.model.Transaction.metadata(Transaction.java:68) ~[IRI.jar:na]
        at com.iota.iri.storage.rocksDB.RocksDBPersistenceProvider.update(RocksDBPersistenceProvider.java:385) ~[IRI.jar:na]
        at com.iota.iri.storage.Tangle.update(Tangle.java:92) ~[IRI.jar:na]
        at com.iota.iri.controllers.TransactionViewModel.update(TransactionViewModel.java:124) ~[IRI.jar:na]
        at com.iota.iri.controllers.TransactionViewModel.setValidity(TransactionViewModel.java:313) ~[IRI.jar:na]
        at com.iota.iri.BundleValidator.validate(BundleValidator.java:105) ~[IRI.jar:na]
        at com.iota.iri.LedgerValidator.getLatestDiff(LedgerValidator.java:84) ~[IRI.jar:na]
        at com.iota.iri.LedgerValidator.updateSnapshot(LedgerValidator.java:249) ~[IRI.jar:na]
        at com.iota.iri.Milestone.updateLatestSolidSubtangleMilestone(Milestone.java:228) ~[IRI.jar:na]
        at com.iota.iri.Milestone.lambda$init$1(Milestone.java:150) ~[IRI.jar:na]
        at java.lang.Thread.run(Thread.java:748) ~[na:1.8.0_131]
```

**Comment by alon-e** [comment]
closing. fixed, merged in 8d1e38b and will be in the next release.
